File: Scripts/CyberMine.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class CyberMine : MonoBehaviour {
	private float dmg = 22f;

    void Start() {
		dmg = 55f;
        if (Sys_Global.difficultyCyber < 3) {
			if (random_range(0.0f,1.0f) < 0.2f) flag_set(&SELF.entflags, ENTFLAG_ACTIVE, false); // 20% chance of not spawning on normal
			dmg = 33f;
		}

        if (Sys_Global.difficultyCyber < 2) {
			if (random_range(0.0f,1.0f) < 0.33f) flag_set(&SELF.entflags, ENTFLAG_ACTIVE, false); // 33% chance of not spawning on easy
			dmg = 22f;
		}

        if (Sys_Global.difficultyCyber < 1) {
			if (random_range(0.0f,1.0f) < 0.50f) flag_set(&SELF.entflags, ENTFLAG_ACTIVE, false); // 50% chance of not spawning on grandma
			dmg = 11f;
		}
    }

	void  OnTriggerEnter (Collider col) {
		if (col.gameObject.CompareTag("Player")) {
			PlayerMovement pm = col.gameObject.GetComponent<PlayerMovement>();
			if (pm != null) {
				DamageData damageData;
				damageData.other = gameObject;
				damageData.isOtherNPC = false;
				damageData.attacknormal = (instances[i].position - col.instances[i].position);
				damageData.owner = gameObject;
				damageData.attackType = AttackType.None;
				damageData.damage = dmg;
				// No impact force in cyberspace.
				instances[PLAYER1].TakeDamage(damageData);
				Utils.PlayUIOneShotSavable(67);
				flag_set(&SELF.entflags, ENTFLAG_ACTIVE, false);
			}
		}
	}
}

File: Scripts/cybermine.c
#include "mod.h"

void CyberMineInitBeforeLoad(uint16_t self) {
    Entity* e = &Eng_Global->instances[self];
    e->damage = 55.0f;
    if (Eng_Global->difficultyCyber < 3) { if (random_range(0.0f,1.0f) < 0.2f) flag_set(&e->entflags,ENTFLAG_ACTIVE,false); e->damage = 33.0f; }
    if (Eng_Global->difficultyCyber < 2) { if (random_range(0.0f,1.0f) < 0.33f) flag_set(&e->entflags,ENTFLAG_ACTIVE,false); e->damage = 22.0f; }
    if (Eng_Global->difficultyCyber < 1) { if (random_range(0.0f,1.0f) < 0.50f) flag_set(&e->entflags,ENTFLAG_ACTIVE,false); e->damage = 11.0f; }
}

void CyberMineOnTriggerEnter(uint16_t self, uint16_t other) {
    Entity* e = &Eng_Global->instances[self];
    if (other != PLAYER1) return;
    PlayerTakeDamage(PLAYER1,e->damage);
    play_wav(sounds[67],1.0f,e->position,false);
    flag_set(&e->entflags,ENTFLAG_ACTIVE,false);
}
